style-guide.md

```diff
@@ -0,0 +1,408 @@
+
+When including different sets of styles for different target viewport sizes using media queries inside the same stylesheet, it is a good idea to make the default styling before any media queries have been applied to the document the narrow screen/mobile styling, and then override this for wider viewports inside successive media queries. [Source](https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Code_guidelines/CSS#use_mobile_first_media_queries).
+
+```
+/*Default CSS layout for narrow screens*/
+
+@media (min-width: 480px) {
+  /*CSS for medium width screens*/
+}
+
+@media (min-width: 800px) {
+  /*CSS for wide screens*/
+}
+
+@media (min-width: 1100px) {
+  /*CSS for really wide screens*/
+}
+```
+
+<!-- TODO: link to mobile breakpoint code paths -->
+
+## CSS
+Guide to terminology: https://github.com/airbnb/css#terminology
+
+### Naming 
+- Prefer dashes over camelCasing in class names (https://github.com/airbnb/css)
+- Prefer dash-cased variable names (e.g. $my-variable) over camelCased or snake_cased variable names.
+- Use human readable selectors that describe what element(s) they style. [Source](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/).
+
+```
+/* Not recommended: meaningless */
+.abc-1901 {}
+
+/* Not recommended: presentational */
+.button-green {}
+.clear {}
+
+/* Recommended: specific */
+.gallery {}
+.login {}
+.video {}
+
+/* Recommended: generic */
+.aux {}
+.alt {}
+```
+
+### Structure
+
+- Refrain from using over-qualified selectors: `div.container` can simply be stated as `.container`
+- Try to avoid using IDs as selectors, as they are overly specific.
+
+### Property ordering
+<!-- TODO: link CSS Wizardry's article on dealing with specificity -->
+Group properties together meaningfully.  Here's a suggested ordering from [Wordpress](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/#property-ordering).
+
+1. Display
+2. Positioning
+3. Box model
+4. Colors and Typography
+5. @include declarations
+6. Nested selectors
+
+### General
+- Write detailed comments for code that isn't self-documenting.  Examples might include: uses of z-index, compatibility or browser-specific hacks
+- Mixins should be used to DRY up your code, add clarity, or abstract complexity--in much the same way as well-named functions.
+- Use flexible/relative units
+- Favor longhand rules over terse shorthand
+
+To start with, it is often harder to understand what the shorthand is doing. It takes a while to pick apart exactly what font syntax is doing.  [Source](https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Code_guidelines/CSS#favor_longhand_rules_over_terse_shorthand).
+
+`font: small-caps bold 2rem/1.5 sans-serif;`
+
+Whereas this is more immediate in terms of understanding:
+```
+font-variant: small-caps;
+font-weight: bold;
+font-size: 2rem;
+line-height: 1.5;
+font-family: sans-serif;
+```
+
+## Javascript
+
```

Comment:
```suggestion

### Use TypeScript

We generally prefer TypeScript over vanilla JavaScript. Occasionally this may not be practical, and that's okay, but as a general rule, if you _can_ use TypeScript, you should.

```